src/module.cpp

```diff
@@ -502,8 +499,10 @@ Expr *lCreateConstExpr(ExprList *exprList, const AtomicType::BasicType basicType
         }
     } else {
         // T equals int8_t* and etc.
+        // We allocate PointToType[N] on the heap. It is managed by unique_ptr.
         using PointToType = typename std::remove_pointer<T>::type;
-        vals = (T)ALLOCA(N * sizeof(PointToType));
+        managedMemory = std::make_unique<PointToType[]>(N);
```

Comment:
I would add some comments here from the discussions https://github.com/ispc/ispc/pull/2788 and https://github.com/ispc/ispc/pull/2739#discussion_r1471150928 to explain why it is implemented this way and what are the problems if you try to implement it "easier".

Reply:
I have added the comment.